bookkeeper-server/src/main/java/org/apache/bookkeeper/client/BKException.java

```diff
@@ -439,12 +439,22 @@ public BKLedgerIdOverflowException() {
      * Extract an exception code from an BKException, or use a default if it's another type.
      */
     public static int getExceptionCode(Throwable t, int defaultCode) {
-        if (t instanceof BKException) {
+        if (t == null) {
+            return BKException.Code.OK;
+        } else if (t instanceof BKException) {
             return ((BKException) t).getCode();
         } else if (t.getCause() != null) {
```

Comment:
not directly related, but I have started working with CompletableFuture in other projects, and I see that dependency on race conditions sometimes CompletableFuture notify CompletionException which wrap the actual exception.
Did you ever seen such problems in BK ?

is this branch addressing such issue ?

Reply:
You'll get CompletionException wrapping, if the future you put your handler is not the future on which completeExceptionally is called.